### PR TITLE
Implement support for building CentOS 7 AMIs

### LINKED ISSUE #79: Add support for CentOS
Allow building of CentOS derived images

### PR DESCRIPTION
**Pull request by gozer**
Fixes #79

### CODE REVIEW

File: bin/generate-base-ami-ids [hunk 1]
```diff
@@ -61,6 +61,12 @@ fi
 
 # Region
 aws_region=$(jq --raw-output '"\(.variables.aws_region)"' < $build_file)
+nubis_account=$(jq --raw-output '"\(.variables.nubis_account_id)"' < $build_file)
+
+if [[ "${nubis_account:-null}" == "null" ]]; then
+  # Official Nubis Marketplace Account ID
+  nubis_account=589768463761
```
Comment (limed) [inline bin/generate-base-ami-ids]: Shouldn't hard code this value, should be a config file I feel. If not in this PR but at least should have an issue opened up to keep track of it

File: bin/generate-base-ami-ids [hunk 2]
```diff
@@ -125,6 +142,16 @@ for builders in $(jq --raw-output '.builders[] | "\(.type)@\([.tags | .platform]
                             --architecture x86_64 \
                             --name 'amzn-ami-hvm-*' || fail "unable to find base ami for $platform on $type"
                ;;
+             centos)
+               aws-find-ami --build-file $build_file \
+                            --output-file $output_file \
+                            --owners 679593333241 \
```
Comment (limed) [inline bin/generate-base-ami-ids]: So the value for line 68 is the the same as this line, does it matter? Whose account ID is this?

Comment (gozer) [inline bin/generate-base-ami-ids]: Yeah, should be cleaned up. It's the account ID for nubis-market, actually.

I'll fix and update the PR

File: bin/generate-base-ami-ids (hunk 1 above)
Comment (gozer) [inline bin/generate-base-ami-ids]: The default is to use the well-known ( sort of ) account ID of nubis-market.

I don't like how Amazon makes these IDs obscure, but there is no way around it.

Better have It once here and not duplicated everywhere, IMO.

**Comment by gozer** [comment]
@limed Fixed your issues

**Pull request merged**